### PR TITLE
feat(pubsub): adding message only when metadataonly=false & adding tests

### PR DESCRIPTION
**Pull request by idonava**

### CODE REVIEW

File: src/events/google_cloud.js [hunk 1]
```diff
@@ -99,14 +100,61 @@ const getMessageData = (reqOptsMessages, index) => {
...
+            return acc;
+        }, []);
+        return parsedMessages;
+    }
+    return null;
+};
+
+const getMessagesFromResponse = (res) => {
+    // eslint-disable-next-line no-debugger
+    debugger;
```
Comment (yotamelkaslasy) [inline src/events/google_cloud.js]: oops?

File: src/events/google_cloud.js [hunk 2]
```diff
@@ -99,14 +100,61 @@ const getMessageData = (reqOptsMessages, index) => {
...
 };
 
+const handlePublishMethod = (messages, config) => {
+    const messageIds = messages && messages.messageIds ? messages.messageIds : messages;
+    if (Array.isArray(messageIds)) {
+        const reqOptsMessages = config.reqOpts && config.reqOpts.messages;
+        const parsedMessages = messageIds.reduce((acc, messageId, currentIndex) => {
+            let message = { id: messageId };
+            // add message only when METADATA_ONLY === FALSE
+            if (!epsagonConfig.getConfig().metadataOnly) {
```
Comment (galbash) [inline src/events/google_cloud.js]: use the third argument to `addMetadata` instead

File: src/events/google_cloud.js [hunk 3]
```diff
@@ -99,14 +100,61 @@ const getMessageData = (reqOptsMessages, index) => {
...
+
+const getMessagesFromResponse = (res) => {
+    // eslint-disable-next-line no-debugger
+    debugger;
+    if (res && res.receivedMessages) {
+        return res.receivedMessages.reduce((acc, current) => {
+            let messageObject = { messageId: current.message.messageId };
+            const messageData = (current.message.data && JSON.parse(`${current.message.data}`));
+            // add message only when METADATA_ONLY === FALSE
+            if (!epsagonConfig.getConfig().metadataOnly) {
```
Comment (galbash) [inline src/events/google_cloud.js]: same

**Review by galbash — CHANGES_REQUESTED** [review]

File: src/events/google_cloud.js (hunk 1 above)
Comment (idonava) [reply src/events/google_cloud.js]: oops.

File: src/events/google_cloud.js (hunk 2 above)
Comment (idonava) [reply src/events/google_cloud.js]: I also wanted to do it, But here I adding the message data to an object who must add to the event.

File: src/events/google_cloud.js (hunk 3 above)
Comment (idonava) [reply src/events/google_cloud.js]: same

**Review by galbash — APPROVED** [review]

**Pull request merged**